### DISCUSSION TITLE
Abandoned lists

### DISCUSSION THREAD

**Discussion by jaschaurbach** [bookwyrm-social/bookwyrm#2483] [Ideas]
### Problem
A USER1 on INSTANCE_A has saved a list from USER2 on INSTANCE_B. Now USER2 deletes the account / INSTANCE_B is getting down. The list is stuck in Limbo - it is there but can not be updated by anyone a it does not belong to anyone. 

Books which are on the list still show that they are on this list as there is a copy of this list on INSTANCE_A, belonging to USER2 (which is not existent any more). 

### How can this be handele at the moment?
At the moment there is nothing anyone can do. 

### Possibilities

There are several possibilities, what can be done software side:

1. Keep status quo.
This would be the easiest thing to do but changes nothing on the problem.

2. Give Admins/Moderators a tool to flag a list as "abandoned". The list gets "(abandoned)" attached to the title. Users on INSTANCE_A can see the list by calling the URL but can not save the list nor is the list shown beside a book. 

3. Give Admins/Moderators a tool to delete a list from the instance when there is no owner for it. "No owner" can be defined as "INSTANCE_B was not reachable for 4 weeks" or "USER2 does not exist for longer than a week"

4. A function for list owners to move ownership to any other user.

5. A function to remove a certain list to show up on book sites.

7. A Function where a user can "claim" a list which is abandoned. this is making a copy of the list and setting the user as owner. After the claim, the list will show that it is based on the list $title by USER2 (because give credits where credits due). If more users are following this list, it is not solving the problem that there is an abandoned list on an instance.

While writing this it just came to my mind that there is no way to _moderate_ a list. I can block a user for the instance and the list does not show up, right. But I can not force a list to be invisible (I like your porn collection but please keep it invisible, thanks). There is no way to restrict users from creating lists (as a warning for people who are in general good persons but do not get the vibe of the instance an need time for that) or a forced rename of a list if it violates the insatnce's CoC.

**Comment by mouse-reeve** [comment]
I think this touches on a bigger issue with federation, which is that any object from a remote instance (a list, a post, a book, et cetera) might have missed an Update, Add, Remove or Delete activity because of transient errors or downtime. So in addition to the case you have with lists, a user profile might be out of date, a status should have been deleted, or a variety of other similar mismatches. 

A fix might be to have an option to update a remote object relative to its origin object (the path for that is stored in the database), and great the remote origin as canonical and update the local object accordingly

**Comment by marcusyoung** [comment]
I have been unable to view a list belong to a user from another instance on my instance and so would never be able to save it. If I open up the profile of a user on another instance within my instance very little is shown. For example, for mouse@bookwyrm.social I see just three books and three status updates. For jascha@books.jascha.wtf (who I know has lists), I see no books, no lists and just a single status. I follow both accounts, and jascha@books.jascha.wtf also follows me. (see screenshots below)

**Reply by mouse-reeve** [reply]
Of interest: #1547

**Comment by marcusyoung** [comment]
![image](https://user-images.githubusercontent.com/10391966/206905452-2c79bd2e-7a4c-4ce8-968a-fda95d705bca.png)



![image](https://user-images.githubusercontent.com/10391966/206905421-943bb5d4-a23c-41dd-913e-0c2795aa7da8.png)
